apps/consumer-client/src/pages/pcdpass-examples/zk-eddsa-event-ticket-proof.tsx

```diff
@@ -0,0 +1,468 @@
+ */
+export function openZKEdDSAEventTicketPopup(
+  urlToPassportWebsite: string,
+  popupUrl: string,
+  fieldsToReveal: EdDSATicketFieldsToReveal,
+  watermark: bigint,
+  validEventIds: string[],
+  externalNullifier?: string
+) {
+  const args: ZKEdDSAEventTicketPCDArgs = {
+    ticket: {
+      argumentType: ArgumentTypeName.PCD,
+      pcdType: EdDSATicketPCDPackage.name,
+      value: undefined,
+      userProvided: true
+    },
+    identity: {
+      argumentType: ArgumentTypeName.PCD,
+      pcdType: SemaphoreIdentityPCDPackage.name,
+      value: undefined,
+      userProvided: true
+    },
+    validEventIds: {
+      argumentType: ArgumentTypeName.StringArray,
+      value: validEventIds.length != 0 ? validEventIds : undefined,
+      userProvided: false
+    },
+    fieldsToReveal: {
+      argumentType: ArgumentTypeName.Object,
+      value: fieldsToReveal,
+      userProvided: false
+    },
+    watermark: {
+      argumentType: ArgumentTypeName.BigInt,
+      value: watermark.toString(),
+      userProvided: false
+    }
+  };
+
+  if (externalNullifier) {
+    args.externalNullifier = {
+      argumentType: ArgumentTypeName.BigInt,
+      value: externalNullifier,
+      userProvided: false
+    };
+  }
+
+  const proofUrl = constructPassportPcdGetRequestUrl<
+    typeof ZKEdDSAEventTicketPCDPackage
+  >(urlToPassportWebsite, popupUrl, ZKEdDSAEventTicketPCDPackage.name, args, {
+    genericProveScreen: true,
+    title: "ZKEdDSA Proof",
+    description: "zkeddsa ticket pcd request"
+  });
+
+  openPassportPopup(popupUrl, proofUrl);
+}
+
+/**
+ * React hook which can be used on 3rd party application websites that
+ * parses and verifies a PCD representing a ZKEdDSA ticket proof.
+ */
+function useZKEdDSAEventTicketProof(
+  pcdStr: string,
+  onVerified: (valid: boolean) => void,
+  fieldsToReveal: EdDSATicketFieldsToReveal,
+  watermark: bigint,
+  externalNullifier?: string
+): { pcd: ZKEdDSAEventTicketPCD | undefined; error: any } {
+  const [error, _setError] = useState<Error | undefined>();
+  const zkEdDSAEventTicketPCD = useSerializedPCD(
+    ZKEdDSAEventTicketPCDPackage,
+    pcdStr
+  );
+
+  useEffect(() => {
+    if (zkEdDSAEventTicketPCD) {
+      verifyProof(
+        zkEdDSAEventTicketPCD,
+        fieldsToReveal,
+        watermark,
+        externalNullifier
+      ).then(onVerified);
+    }
+  }, [
+    zkEdDSAEventTicketPCD,
+    fieldsToReveal,
+    watermark,
+    externalNullifier,
+    onVerified
+  ]);
+
+  return {
+    pcd: zkEdDSAEventTicketPCD,
+    error
+  };
+}
+
+async function verifyProof(
+  pcd: ZKEdDSAEventTicketPCD,
+  fieldsToReveal: EdDSATicketFieldsToReveal,
+  watermark: bigint,
+  externalNullifier?: string
+): Promise<boolean> {
+  // TODO: after eddsa keys are represented in normal form and not montgomery
+  // form, we can remove the call to Package initialization logic and also
+  // remove the package artifacts in the ../../public directory (for now)
```

Comment:
this should be done this week: #462. I recommend creating an issue to follow up on this TODO and placing it on next week's milestone.